### ISSUE TITLE
IOTManager fails when protocol is empty

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/iotagent-manager#142]
We have found that IOTManager fails when protocol is empty.

```
POST /iot/devices
{
  "devices": [
    {
      ...
      "protocol": ""
    }
  ]
}
```

In this case, the response should be either 404 Not Found (which is correctly generated when protocol is unkwnon but filled, e.g. `"protocol": "IoT-xx"`) or 400 Bad Request.

**Comment by AlvaroVega** [comment]
Any error in json body produces a 500 error also!

**Comment by fgalan** [comment]
Missing protocol bug fixed in PR #143. Error 500 on JSON syntax spinned off to separate issue #144
